### DISCUSSION TITLE
Clarification on TLS Support with NXP Plug-and-Trust Provider for OpenSSL 3.0

### DISCUSSION THREAD

**Discussion by richardh538** [openssl/openssl#25949] [Q&A] [answered]
Hello,

I am looking for clarification regarding the use of the NXP Plug-and-Trust provider with OpenSSL 3.0. Specifically, does the provider interface itself support TLS? According to [NXP](https://community.nxp.com/t5/Other-NXP-Products/SE050-OpenSSL-3-Provider-configuration/m-p/1725113/highlight/true#M19298), it seems that it does not work for TLS.

Is this limitation inherent to the provider interface, or is it specific to NXP's [EdgeLock SE05x Plug & Trust Middleware](https://www.nxp.com/products/security-and-authentication/authentication/edgelock-se050-plug-and-trust-secure-element-family-enhanced-iot-security-with-high-flexibility:SE050#software)?

Thank you in advance for any clarification.

Thanks
/Richard

**Comment by t8m** [comment, marked as answer]
The provider interface is fully capable of supporting TLS - all the crypto used by default by the TLS implementation in libssl is using the crypto implementations from the default provider through the provider interface.

Of course third party providers may not support all the required functionality. In most cases it is possible however to configure the providers and the default property query in such way that the third party provider is used just for the functionality it provides and the missing functionality will come from the default provider.
